packages/expo-localization/ios/LocalizationModule.swift

```diff
@@ -15,16 +15,28 @@ public class LocalizationModule: Module {
       return Self.getCurrentLocalization()
     }
   }
+  
+  // If the application isn't manually localized for the device language then the
+  // native `Locale.current` will fallback on using English US
+  // [cite](https://stackoverflow.com/questions/48136456/locale-current-reporting-wrong-language-on-device).
+  // This method will attempt to return the locale that the device is using regardless of the app,
+  // providing better parity across platforms.
+  static func getPreferredLocale() -> Locale {
+      guard let preferredIdentifier = Locale.preferredLanguages.first else {
+          return Locale.current
+      }
+      return Locale(identifier: preferredIdentifier)
+  }
 
   static func getCurrentLocalization() -> [String: Any?] {
-    let locale = Locale.current
+    let locale = getPreferredLocale()
     let languageCode = locale.languageCode ?? "en"
     var languageIds = Locale.preferredLanguages
 
     if languageIds.isEmpty {
       languageIds.append("en-US")
     }
-
+    
```

Comment:
```suggestion

```

I recommend turning on this option in Xcode 👇 

![Screen Shot 2022-01-06 at 14 27 21](https://user-images.githubusercontent.com/1714764/148390363-1ff0b48b-9374-44c2-91d8-a8435a66ec1b.png)